pkg/resource/deploy/step_generator.go

```diff
@@ -305,6 +306,8 @@ func (sg *stepGenerator) generateSteps(event RegisterResourceEvent) ([]Step, res
 
 				// register the alias with the provider registry
 				sg.deployment.providers.RegisterAlias(urn, urnOrAlias)
+
+				alias = []resource.URN{urnOrAlias}
```

Comment:
Nit: Worth a comment here describing why we're only saving the single alias?

Reply:
Done.